lib/utils.pm

```diff
@@ -1658,6 +1659,53 @@ sub script_output_retry {
     die("Waiting for Godot: $cmd") if $die;
 }
 
+
+=head2 validate_script_output_retry
+
+ validate_script_output_retry($cmd, $check, [retry => $retry], [delay => $delay], [timeout => $timeout]);
+
+Repeat command until validate_script_output succeeds or die. Return the output of the command on success.
+
+C<$retry> refers to the number of retries and defaults to C<10>.
+
+C<$delay> is the time between retries and defaults to C<30>.
+
+If the command doesn't succeed after C<$retry> retries,
+this function will die.
+
+Example:
+
+ validate_script_output_retry('ping -c1 -W1 machine', m/1 packets transmitted/, retry => 5, delay => 60);
+
+=cut
+sub validate_script_output_retry {
+    my ($cmd, $check, %args) = @_;
+    $args{retry} //= 10;
+    $args{delay} //= 30;
+    $args{timeout} //= 90;
+    my $retry = delete $args{retry};
+    my $delay = delete $args{delay};
+    my $ret;
+
+    my $exec = "timeout $args{timeout} $cmd";
+    # Exclamation mark needs to be moved before the timeout command, if present
+    if (substr($cmd, 0, 1) eq "!") {
+        $cmd = substr($cmd, 1);
+        $cmd =~ s/^\s+//;    # left trim spaces after the exclamation mark
+        $exec = "! timeout $args{timeout} $cmd";
+    }
+
+    $args{timeout} += 3;    # timeout for script_run must be larger than for the 'timeout ...' command
+    $args{procced_on_failure} = 1;
```

Comment:
```suggestion
    $args{proceed_on_failure} = 1;
```
But I would let this up to the caller! SOrry I over seen this on my review!

Reply:
You don't need to be sorry, it's my fault I type too quick :D 
Good that you spotted it, thank you!